src/main/java/gmbs/game/BaseballRuleChecker.java

```diff
@@ -0,0 +1,58 @@
+package gmbs.game;
+
+import gmbs.common.Constant;
+
+import java.util.List;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+public class BaseballRuleChecker {
+
+    private static final int START_INDEX = 0;
+
+    private final int strikeCount;
+    private final int ballCount;
+
+    public BaseballRuleChecker(List<Integer> randomNumberList, List<Integer> inputNumberList) {
+        this.strikeCount = setStrikeCount(randomNumberList, inputNumberList);
+        this.ballCount = setBallCount(randomNumberList, inputNumberList);
+    }
+
+    private int setStrikeCount(List<Integer> randomNumberList, List<Integer> inputNumberList) {
+         return (int) Stream.iterate(START_INDEX, increaseOneOperator()).limit(Constant.MAX_INPUT_SIZE)
+                .filter(index -> isSameValueAtSameIndex(randomNumberList, inputNumberList, index))
+                .count();
```

Comment:
```suggestion
        return (int) IntStream.range(START_INDEX, Constant.MAX_INPUT_SIZE)
                .filter(index -> isSameValueAtSameIndex(randomNumberList, inputNumberList, index))
                .count();
```
IntStream을 사용하는 건 어떨까요~?

Reply:
IntStream 사용을 하는게 큰 차이가 있을까요??

Reply:
increaseOneOperator를 생략할 수 있어서 더 간결하게 가져갈 수 있을 것 같습니다~
그리고 Stream을 통해서 Integer, Long, Double 등의 타입을 많이 사용해서 IntStream, LongStream, DoubleStream 등이 나온 걸로 알고 있습니다~!